env/types.go

```diff
@@ -126,23 +143,33 @@ func ValidIdentifier(n string) error {
 
 // Set sets a variable.
 func Set(name, value string) error {
-	err := ValidIdentifier(name)
-	if err != nil {
+	if err := ValidIdentifier(name); err != nil {
 		return err
 	}
-	switch name {
-	case "QUIET":
-		if value == "" {
-			value = "on"
-		} else {
-			value, err = ParseBool(value, name)
-			if err != nil {
-				return err
-			}
+	// validate the variable value if there's a validation funciton defined in variableValidationFunctions before setting
+	if valueIsValid, ok := variableValidationFunctions[name]; ok {
+		if !valueIsValid(value) {
+			return fmt.Errorf(text.FormatFieldInvalid, value, name)
 		}
 	}
-	vars.Set(name, value)
-	return nil
+	// QUIET accepts special values
+	// other vars are a simple set
+	if name != "QUIET" {
```

Comment:
I'm not sure replacing `switch` makes it more readable. Maybe there should be a map of variable value parser functions, not just validators? Then you could move the special case for `QUIET` into a dedicated function.

Reply:
A switch for one case seemed redundant if the control flow just handles the non-QUIET-case first. Related to the other comment though ON_ERROR_STOP could be adjusted to behave consistent with QUIET which might be better.